src/core/library.c

```diff
@@ -2556,3 +2619,48 @@ MsQuicExecutionPoll(
 }
 
 #endif
+
+_IRQL_requires_max_(PASSIVE_LEVEL)
+QUIC_STATUS
+QuicLibrarySetRetryKeyConfig(
+    _In_ const QUIC_STATELESS_RETRY_CONFIG* Config
+    )
+{
+    if (Config->Algorithm > QUIC_AEAD_ALGORITHM_CHACHA20_POLY1305) {
+        QuicTraceLogError(
+            LibrarySetRetryKeyAlgorithmInvalid,
+            "[ lib] Invalid retry key algorithm: %d.",
+            Config->Algorithm);
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+    if (Config->RotationMs == 0) {
+        QuicTraceLogError(
+            LibrarySetRetryKeyRotationInvalid,
+            "[ lib] Invalid retry key rotation ms: %u.",
+            Config->RotationMs);
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    if (Config->Secret == NULL) {
+        QuicTraceLogError(
+            LibrarySetRetryKeySecretNull,
+            "[ lib] Invalid retry key secret: NULL.");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+    if (Config->SecretLength < CxPlatKeyLength((CXPLAT_AEAD_TYPE)Config->Algorithm)) {
+        QuicTraceLogError(
+            LibrarySetRetryKeySecretLengthInvalid,
+            "[ lib] Invalid retry key secret length: %u. Expected %u.",
+            Config->SecretLength,
+            CxPlatKeyLength((CXPLAT_AEAD_TYPE)Config->Algorithm));
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+    size_t SecretLen = CXPLAT_MIN(sizeof(MsQuicLib.BaseRetrySecret), Config->SecretLength);
```

Comment:
Shouldn't this just be `CxPlatKeyLength((CXPLAT_AEAD_TYPE)Config->Algorithm)`?

Reply:
I was thinking of the future change where we use a different key derivation algorithm. But I'll just update all this for that PR when it happens and use `!=` here.